orchestrator/services/tasks.py

```diff
@@ -0,0 +1,118 @@
+
+
+def get_celery_task(task_name: str) -> Task:
+    if _celery:
+        return _celery.signature(task_name)
+    else:
+        raise AssertionError("Celery has not been initialised yet")
+
+
+def register_custom_serializer() -> None:
+    # surf specific serializer to correctly handle more complex classes
+    registry.register("surf", json_dumps, json_loads, "application/json", "utf-8")
+
+
+def initialise_celery(celery: Celery) -> None:
+    global _celery
+    if _celery:
+        raise AssertionError("You can only initialise Celery once")
+    _celery = celery
+
+    # Different routes/queues so we can assign them priorities
+    celery.conf.task_routes = {
+        NEW_TASK: {"queue": "tasks"},
+        NEW_WORKFLOW: {"queue": "workflows"},
+        RESUME_TASK: {"queue": "tasks"},
+        RESUME_WORKFLOW: {"queue": "workflows"},
+    }
+
+    register_custom_serializer()
+
+    def start_process(pid: UUID, workflow_key: str, state: Dict[str, Any], user: str) -> Optional[UUID]:
+        try:
+            workflow = get_workflow(workflow_key)
+
+            if not workflow:
+                raise_status(HTTPStatus.NOT_FOUND, "Workflow does not exist")
+
+            pstat = ProcessStat(pid, workflow=workflow, state=Success(state), log=workflow.steps, current_user=user)
+
+            safe_logstep_with_func = partial(safe_logstep, broadcast_func=None)
+            pid, _ = _run_process_async(pstat.pid, lambda: runwf(pstat, safe_logstep_with_func))
+
+        except Exception as exc:
+            local_logger.error("Worker failed to execute workflow", pid=pid, details=str(exc))
+            return None
+        else:
+            return pid
+
+    def resume_process(pid: UUID, user_inputs: Optional[List[State]], user: str) -> Optional[UUID]:
+        try:
+            process = _get_process(pid)
+            pid, _ = thread_resume_process(process, user_inputs=user_inputs, user=user)
+        except Exception as exc:
+            local_logger.error("Worker failed to resume workflow", pid=pid, details=str(exc))
+            return None
+        else:
+            return pid
+
+    @celery.task(log=local_logger, name=NEW_TASK, serializer="surf")  # type: ignore
+    def new_task(pid, workflow_key: str, state: Dict[str, Any], user: str) -> Optional[UUID]:
+        local_logger.warning("Start task", workflow_key=workflow_key)
+        return start_process(pid, workflow_key, state=state, user=user)
+
+    @celery.task(log=local_logger, name=NEW_WORKFLOW, serializer="surf")  # type: ignore
+    def new_workflow(pid, workflow_key: str, state: Dict[str, Any], user: str) -> Optional[UUID]:
+        local_logger.info("Start workflow", workflow_key=workflow_key)
+        return start_process(pid, workflow_key, state=state, user=user)
+
+    @celery.task(log=local_logger, name=RESUME_TASK)  # type: ignore
+    def resume_task(pid: UUID, user_inputs: Optional[List[State]], user: str) -> Optional[UUID]:
+        local_logger.warning("Resume task")
```

Comment:
Why is this a "warning" with no other logging? Perhaps at least include the pid as key value in the log statement. This goes for all the @celery.task decorated funcs.

Reply:
Should be info/debug. Was a leftover from local debugging/testing that makes warnings easier to spot.